### ISSUE TITLE
Adapting URI prefix from foreign attributes, questions, options... does not allow parallel use

### ISSUE THREAD

**Issue by schmitzd** [rdmorganiser/rdmo#223]
At the documentation I found the following hints:

> When you import a question catalog or any other content from another institution these imports do have a URI prefix different from yours. If you change elements from these third party imports we strongly recommend to always adjust the URI prefix into your own one to make the changes persistent. This is necessary because a re-import of the third party content will overwrite by using the URI as identifier. Data in your database having the same URI as the imported ones will get updated and so overwritten. Please do also look into Export and Import page for a little more detail.
(from https://rdmo.readthedocs.io/en/latest/management/index.html )
and also
> A Project import will always create a new Project holding the imported data. Note that for all other types (Options, Tasks, etc.) data already existing in your RDMO installation will be overwritten during an import. The URI acts as identifier. If you import an element that equals the URI of an already existing one, the import will overwrite.
(from https://rdmo.readthedocs.io/en/latest/management/export.html)

In our case we (too late) learned that we should use a different (our) URI prefix and wanted to change that by reloading an XML file where this was corrected.

It turned out that - despite what one could expect from above - the *new* options/optionsset (because they have a different URI and URI prefix) were **not** imported. 

By looking at the code, I presume that the reason is the *validation*. While the URI is obviously different, the key validation (for optionsets) and the path validation (for options) only look at the uniqueness of keys and paths - **not including the prefix or the URI**.

In conclusion, it does not seem possible to have in general any attribute, question, option, optionset etc. in parallel to a similar item where *only* the URI prefix is changed. At least, you will not be able to import such files due to the validation.

Could you look into this? I did not dive any deeper to find out whether it is sufficient to extend the validation to the uri prefix or whether at some other part of the implementation you indeed rely on the uniquess of key or path only.

Thanks in advance!

**Comment by jochenklar** [comment]
Hi, sorry for the late reply. This is correct, the path/key needs to be different for every element (e.g. attribute, questions, ...). By setting a different URI prefix, you prevent a subsequent import to overwrite your changes to an element. I just refactored the Import quiet a bit so that it is more visible what is imported and what is not.

**Comment by jochenklar** [comment]
After giving it some thought ... maybe relying on the uniqueness of the path doesn't make much sense anymore, since the relations are resolved using the URI when importing anyway. @triole we need to talk about this :smile:

**Comment by jochenklar** [comment]
This issue was solved for RDMO 1.4. Pathes do not need to be unique anymore, URI are.
